Build also as a shared library and add install target

jirutka opened the pull request:
BTW: I’ve packaged [libpg_query](https://pkgs.alpinelinux.org/package/edge/testing/x86_64/libpg_query) and [ruby-pg_query](https://pkgs.alpinelinux.org/package/edge/testing/x86_64/ruby-pg_query) for Alpine Linux. It’s built with patches from this PR and few others to unbundle dependencies.

jirutka committed:
Build also as a shared library

jirutka committed:
Add install target to Makefile

lfittl commented on Makefile:
```diff
@@ -251,3 +267,17 @@ test/scan: test/scan.c test/scan_tests.c $(ARLIB)
+install: $(ARLIB) $(SOLIB)
+	$(INSTALL) -d "$(DESTDIR)"$(libdir)
+	$(INSTALL) -m 644 $(ARLIB) "$(DESTDIR)"$(libdir)/$(ARLIB)
+	$(INSTALL) -m 755 $(SOLIB) "$(DESTDIR)"$(libdir)/$(SOLIBVER)
```
Does the shared library actually need to be executable? (I would have thought 644 is sufficient here)

jirutka replied:
Yes, shared libraries are always executable, just look into your `/usr/lib` (or `/usr/lib64`)…

lfittl commented on Makefile:
```diff
@@ -1,13 +1,24 @@
+VERSION_MINOR = $(call word-dot,$(VERSION),2)
+VERSION_PATCH = $(call word-dot,$(VERSION),3)
+
+SONAME = $(SOLIB).$(shell printf '%02d%02d' $(PG_VERSION_MAJOR) $(VERSION_MAJOR))
```
I think we should include the minor version into the SONAME, since we may have ABI breaks with minor releases as well (but it seems reasonable to avoid them with patch releases).

jirutka replied:
Resolved by ade04b03f1e3955558d9489a0e1b5604c0210802.

lfittl commented on Makefile:
```diff
@@ -74,14 +87,14 @@ endif
 all: examples test build
 
-build: $(ARLIB)
+build: $(ARLIB) $(SOLIB)
```
I'd like to avoid building the shared library by default, since its a special case - the default expectation is that pg_query is used as a static library, and examples/etc are written with that in mind.

I'd recommend we make a separate `build_shared` (or similar) target, and install depends on that as well as `build`.

jirutka replied:
Resolved by 931a0b5038e2f9d869d7989c53b21f09509e6e3e.

lfittl reviewed (commented):
@jirutka Thanks for the contribution!

Whilst I'm not sure if its a good idea to use libpg_query as a shared library, and it was written with a static library in mind, it doesn't hurt either I think.

My only requests here are to avoid building the shared library by default, and to include the minor version in the SONAME (since it's likely we'll break the ABI in a minor release sometimes).

jirutka committed:
Don't build shared lib by default

Requested in https://github.com/pganalyze/libpg_query/pull/100#discussion_r628403551

jirutka committed:
Include minor version in SONAME

Requested in https://github.com/pganalyze/libpg_query/pull/100#discussion_r628402291

jirutka commented:
I made the requested changes. Could you please take a second look?
